btcutil/psbt/finalizer.go

```diff
@@ -519,6 +520,15 @@ func finalizeTaprootInput(p *Packet, inIndex int) error {
 	switch {
 	// Key spend path.
 	case len(pInput.TaprootKeySpendSig) > 0:
+		// Make sure TaprootKeySpendSig is equal to size of signature, if not,
+		// we assume that sighash flag was appended to the signature.
+		if len(pInput.TaprootKeySpendSig) == schnorr.SignatureSize {
+			// Append to the signature if flag is not equal to Default.
+			if pInput.SighashType != txscript.SigHashDefault {
+				sigHashType := byte(pInput.SighashType)
+				pInput.TaprootKeySpendSig = append(pInput.TaprootKeySpendSig, sigHashType)
```

Comment:
I think we shouldn't modify the packet itself.
How about this?
```
		sig := pInput.TaprootKeySpendSig

		// Make sure TaprootKeySpendSig is equal to size of signature,
		// if not, we assume that sighash flag was appended to the
		// signature.
		if len(pInput.TaprootKeySpendSig) == schnorr.SignatureSize {
			// Append to the signature if flag is not equal to the
			// default sighash (that can be omitted).
			if pInput.SighashType != txscript.SigHashDefault {
				sigHashType := byte(pInput.SighashType)
				sig = append(sig, sigHashType)
			}
		}
		serializedWitness, err = writeWitness(sig)
```

Reply:
Of course, it's very important for readability.